### PR TITLE
Automatic naming should never use Contact name for non-HH accounts

### PR DESCRIPTION
**Pull request by npsp-reedestockton**
# Critical Changes

# Changes
- We fixed an issue in Enhanced Recurring Donations automatic naming that caused Recurring Donations with both a Contact and an Account specified to use Contact name instead of Account name in the naming formula when the Account is a non-Household Account.

# Issues Closed
- Issue #5514

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by npsp-reedestockton** [comment]
**lurch: attach W-041644

**Comment by npsp-reedestockton** [comment]
@force2b Ready for review. Thanks!
@sam-knox Please review PR description. No label changes. Thanks!

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000007hajeQAA" target="_blank">W-041644</a>

File: src/classes/RD2_Donor.cls [hunk 1]
```diff
@@ -49,6 +49,7 @@ public with sharing class RD2_Donor {
     public class Record {
         public Id recordId { get; private set; }
         public String name { get; private set; }
+        public String recordTypeId { get; private set; }
```
[the remark this answers is not in the document]
Comment (npsp-reedestockton) [reply src/classes/RD2_Donor.cls]: Yes, definitely.  Thanks for that!

**Pull request merged**